plugins/inputs/supervisor/supervisor.go

```diff
@@ -0,0 +1,170 @@
+package supervisor
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/kolo/xmlrpc"
+	"net/url"
+)
+
+type Supervisor struct {
+	Log telegraf.Logger `toml:"-"`
+
+	Server         string `toml:"url"`
+	PidGather      bool   `toml:"gather_pid"`
+	ExitCodeGather bool   `toml:"gather_exit_code"`
+	UseIdentTag    bool   `toml:"use_identification_tag"`
+
+	Status SupervisorInfo
```

Comment:
Do we really need this as an struct member or can we use a local variable in the `Gather` function? If we _really_ need to keep it, at least do not export it.

Reply:
I choose to make this variable a struct member for easy access to it in functions, i'll make it private for now.

Reply:
To be honest, passing those down the few functions that need it would be much better compared to adding an unnecessary _state_ to the plugin.